File: Go/Function/1065.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func solve(a int) int {
	if a <= 99 {
		return a
	} else {
		count := 99
		for i := a; i > 99; i-- {
			th := i / 1000
			h := (i % 1000) / 100
			t := (i % 100) / 10
			o := (i % 10)
			if th == 0 {
				if (h - t) == (t - o) {
					count++
				} else {
					continue
				}
			}
		}
		return count
	}
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	var input int
	fmt.Fscan(reader, &input)
	writer := bufio.NewWriter(os.Stdout)
	fmt.Fprint(writer, solve(input))

	defer writer.Flush()

}

File: python/Function/1065.py
import sys


def solve(a):
    if(a <= 99):
        return a
    else:
        count = 99
        for i in range(a, 99, -1):
            th = i//1000
            h = (i % 1000)//100
            t = (i % 100)//10
            o = (i % 10)
            if(th == 0):
                if((h-t) == (t-o)):
                    count += 1
            else:
                continue
        return count


if __name__ == "__main__":
    num = int(sys.stdin.readline())
    print(solve(num))
